sizefashion/src/pages/SingleProductPage/SingleProductPage.tsx

```diff
@@ -0,0 +1,49 @@
+import { useEffect } from 'react'
+import { useParams } from 'react-router-dom'
+import { useProductsContext } from '../../context/products_context'
+import { ProductImages, Loading } from '../../components'
+import { BackToProductsButton } from './BackToProductsButton'
+import { SingleProductContent } from './SingleProductContent'
+import ErrorPage from '../ErrorPage'
+
+const SingleProductPage = () => {
+  const { slug } = useParams<{ slug: string }>()
+  const {
+    singleProduct,
+    fetchSingleProduct,
+    singleProductLoading,
+    singleProductError,
+    allProducts,
+  } = useProductsContext()
+
+  const { image } = { ...singleProduct }
+
+  // caso aconteça um refresh na tela de produto
+  useEffect(() => {
+    if (slug) {
+      fetchSingleProduct(slug)
+    }
+    // eslint-disable-next-line
+  }, [slug, allProducts])
+
+  if (singleProductLoading) {
+    return <Loading />
+  }
+  if (singleProductError) {
+    return <ErrorPage />
+  } else {
```

Comment:
@otaaviostein no need to wrap your outer code, since your two previous conditions are already returning something! We can safely remove the `else` block and let the code within outside.